backend/corpora/lambdas/api/v1/collection.py

```diff
@@ -1,16 +1,16 @@
-from flask import make_response, jsonify
 from typing import Optional
 
+from flask import make_response, jsonify, g
+
 from ....common.corpora_orm import DbCollection, CollectionVisibility
-from ....common.utils.db_utils import db_session
 from ....common.entities import Collection
 from ....common.utils.exceptions import ForbiddenHTTPException
 
 
-@db_session()
 def get_collections_list(from_date: int = None, to_date: int = None, user: Optional[str] = None):
-
+    session = g.db
```

Comment:
instead of importing g from flask what if you created a get_db function and added it to the app?
Something like
```
def get_db():
    if 'db' not in g:
        g.db = connect_to_database()

    return g.db
```
https://flask.palletsprojects.com/en/1.1.x/appcontext/

Reply:
I saw that, and did that for a while. I decided to remove it because it added another path for managing a session other than the existing `db_session_manager`. Using the `existing db_session_manager` adds less redundant code to the project. I also couldn't decide where in the project a `get_db` function should live.

Reply:
I think it would be ok to put it in app.py

I think adding it as a function will improve readability. Right now g.db is all over the codebase and I think its not immediate clear what it is doing. Because its in so many places I think it makes more sense to rename it than to add comments about what it is doing

Reply:
it can't live in app, unless we put in where `db_session_manager` is. `flask.g` only exist within the context of `flask_app.test_request_context`.

Reply:
discussed further over slack. Summary -- its not ideal but its the best of bad options. In the future we may refactor to use flask-sqlalchemy or fast api